src/lib/eslint/BaseEslintEngine.ts

```diff
@@ -230,19 +236,21 @@ export abstract class BaseEslintEngine implements RuleEngine {
 	}
 
 	/* eslint-disable @typescript-eslint/no-explicit-any */
-	private selectRelevantRules(rules: Rule[]): Record<string,any> {
+	private async selectRelevantRules(rules: Rule[]): Promise<Record<string,any>> {
 		const filteredRules = {};
-		let ruleCount = 0;
-
-		for (const rule of rules) {
-			// Find if a rule is relevant
-			if (rule.engine === this.getName()) {
-				// Select rules by setting them to "error" level in eslint config
-				filteredRules[rule.name] = "error";
-				ruleCount++;
+
+		// the eslint engines will run all rules explicitly enabled and all 'recommended' rules
+		// that are inherited from the 'extends' configuration attribute. Disable all rules that
+		// the engine would run by default. The requested rules will be  enabled below.
+		if (rules.length > 0) {
+			for (const rule of (await this.getCatalog()).rules.filter(r => r.defaultEnabled)) {
+				filteredRules[rule.name] = 'off';
 			}
 		}
-		this.logger.trace(`Count of rules selected for ${this.getName()}: ${ruleCount}`);
+
+		rules.forEach(rule => filteredRules[rule.name] = 'error');
+
+		this.logger.trace(`Count of rules selected for ${this.getName()}: ${rules.length}`);
```

Comment:
This message is no longer accurate, right? Shouldn't it be the number of rules in`filteredRules` that don't have a value of `off`?

Reply:
Ah. I must have misunderstood the code. You're turning everything to off, and then using `rules` to turn on those specific rules. That makes sense.